### ISSUE TITLE
Device Handler Error

### ISSUE THREAD

**Issue by ndmax** [yracine/device-type.myecobee#26]
When following the provided instructions to create the ST device handler from the code provided in `MyEcobeeDevice.txt` I receive the following error:

`Org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed: script_dth_metadata_7d75c11f_f985_4dd1_a10f_bb32bbba35c5: 1: Invalid variable name. Must start with a letter but was: ﻿ . At [1:1] @ line 1, column 1. ﻿/*** ^ 1 error`

After dropping $20 on the code, I'd like to get it working as soon as possible.  Any suggestions?

**Comment by ndmax** [comment]
I can add the Init app, but see the following error in ST Live Logging: 

`physicalgraph.app.exception.UnknownDeviceTypeException: Device type 'My Ecobee Device' in namespace 'yracine' not found. @line 703 (doCall)`

...which I assume is due to the missing device handler.

**Comment by yracine** [comment]
Hi, it seems that you haven't followed the directions in the README at my github, specifically the ones in the prerequisites section about installing the code in the right shard.  Also, don't forget to SAVE & PUBLISH the code in the IDE; otherwise, the code for the DTH will not be found.
  
Please follow the README instructions as it was tested & executed by thousands of ST users.

I'm closing this issue as the installation has been executed by many contributors without issues.

**Comment by ndmax** [comment]
Thanks @yracine for taking the time to reply.  I'm afraid it isn't accurate.

I am using the correct shard.  I did publish the Init app code, but of course it was impossible to publish the device handler code since the error occurred when trying to "create" the device handler (from code) - a step required prior to publishing.  It's a subjective statement, but I do believe I read the instructions thoroughly.

After reviewing all troubleshooting steps, I finally noticed that the error message was referencing 1:1 (first line, first column), and citing `/**` as the trigger for the error.  So, I removed the comments at the beginning of the device handler code.  The device handler installed normally, and from that point I was able to get the full device integration working.

Since I was able to discover a workaround (and the integration works well), the error is no longer an issue for me.  It might be of interest to you if some external process has changed - such as how ST imports code from the API dashboard - which could complicate new installations regardless of how many successful executions have been completed previously without issue.

**Comment by yracine** [comment]
Hi, I'm sorry that you had some kind of issues with the _copy and paste_ of the code, but the code per se is not at fault here. 

I was able to create the DTH without any issue without modifying the code as it was downloaded to you.

You may have used a tablet  to do the _copy and paste_ or inserted some extra comments during the operation.

The important thing is that it works for you now.

Regards.

**Comment by ndmax** [comment]
Thanks, @yracine, I probably should have seen a response like that coming.  I _copied and pasted_ using Atom (1.45.0) and Chrome (80.0.3987.149) on Ubuntu (19.10 x64).  I alternated with Gedit and FF as well with the same result. I am familiar with the copy-paste process.

So, I did a little more investigating.  It turns out that the first character in your device handler code is a Zero Width No-Break Space (U+FEFF, a.k.a. Byte-Order-Mark).

Most likely, the space was introduce _when you copied and pasted_ the code from your editor into the text file.  Some editors (such as Notepad on Windows) will always add a BOM when you save a file with the UTF-8 encoding.  Both Gedit and Atom pass the BOM character into the ST API dashboard, which ST fails to interpret, resulting in the error.  The BOM character does not appear in your Init app code, which installs successfully.

It might be useful to modify `MyEcobeeInit.txt` with an editor capable of displaying non-printable unicode characters and removing the U+FEFF.  That hidden space will break your device for others, even if it works for you now.
